File: Examples/test-suite/ruby/using_private_runme.rb
require 'using_private'

include Using_private

f = FooBar.new
f.x = 3

if f.blah(4) != 4
  raise RuntimeError, "blah(int)"
end

if f.defaulted() != -1
  raise RuntimeError, "defaulted()"
end

if f.defaulted(222) != 222
  raise RuntimeError, "defaulted(hi)"
end

File: Examples/test-suite/lua/using_private_runme.lua
require("using_private")

local f = using_private.FooBar()
f.x = 3
assert(f:blah(4) == 4, "blah(int)")
assert(f:defaulted() == -1, "defaulted()")
assert(f:defaulted(222) == 222, "defaulted(hi)")
